src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts

```diff
@@ -14,77 +14,77 @@ import {
   ProceedingsOrderTypes,
   Respondent,
 } from '../../../app/case/definition';
-import { AppRequest } from '../../../app/controller/AppRequest';
 
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-export const setProceedingDetails = (UserCase, respondent: Respondent, req: AppRequest): Respondent => {
-  const respondentDetails = respondent;
+export const setProceedingDetails = (userCase: CaseWithId): CurrentOrPreviousProceedings => {
   const currentOrPreviousProceedings: CurrentOrPreviousProceedings = {};
   const proceedingDetails: ProceedingDetailsData[] = [];
+  const { proceedingsStart, proceedingsStartOrder, courtProceedingsOrders, otherProceedings } = userCase;
 
-  if (req.session.userCase.courtProceedingsOrders) {
-    UserCase['courtProceedingsOrders'].forEach(order => {
-      if (UserCase['otherProceedings']?.['order'].hasOwnProperty(`${order}s`)) {
+  if (courtProceedingsOrders) {
+    courtProceedingsOrders.forEach(order => {
+      if (
```

Comment:
Why is this condition required?

Reply:
Basically that one is having userCase.courtProceedingOrders... like that.. I used const {...} and destructured those values... so using them as variables like... courtProceedings instead of userCase.courtProceedings.